sanitize/sanitize_plan.go

```diff
@@ -47,96 +47,60 @@ func SanitizePlan(old *tfjson.Plan) (*tfjson.Plan, error) {
 // the BeforeSensitive and AfterSensitive in outputs are opaquely the
 // same.
 //
-// Sensitive values are replaced with the value supplied with
-// replaceWith. A copy of the Plan is returned.
-func SanitizePlanWithValue(old *tfjson.Plan, replaceWith interface{}) (*tfjson.Plan, error) {
-	if old == nil {
-		return nil, NilPlanError
-	}
-
-	result, err := copyPlan(old)
-	if err != nil {
-		return nil, err
+// Sensitive values are replaced with the value supplied with replaceWith.
+func SanitizePlanWithValue(result *tfjson.Plan, replaceWith interface{}) error {
+	if result == nil {
+		return NilPlanError
 	}
```

Comment:
I know this was already in place but I think we can also remove this check because at least the CLI calls `plan.Validate()` before doing the sanitize.